pkg/clustermesh/clustermesh.go

```diff
@@ -209,18 +208,22 @@ func NewClusterMesh(c Configuration) (*ClusterMesh, error) {
 			Name:      "remote_cluster_nodes",
 			Help:      "The total number of nodes in the remote cluster",
 		}, []string{metrics.LabelSourceCluster, metrics.LabelSourceNodeName, metrics.LabelTargetCluster}),
-		ipcache: c.IPCache,
 	}
 
-	w, err := createConfigDirectoryWatcher(c.ConfigDirectory, cm)
+	lifecycle.Append(cm)
+	return cm
+}
+
+func (cm *ClusterMesh) Start(hive.HookContext) error {
```

Comment:
Ha, interesting! I kind of prefer the "cm implements a startable interface" rather than "cm appends hooks". I didn't realise how easy it is to convert from one to the other.

Reply:
I'm not sure I fully got your comment. Are you suggesting to adopt a different approach?